pkg/apis/elasticsearch/v1/elasticsearch_types.go

```diff
@@ -72,8 +77,175 @@ type ElasticsearchSpec struct {
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:validation:Enum=DeleteOnScaledownOnly;DeleteOnScaledownAndClusterDeletion
 	VolumeClaimDeletePolicy VolumeClaimDeletePolicy `json:"volumeClaimDeletePolicy,omitempty"`
+
+	// Monitoring enables you to extract log and stack monitoring metrics of this Elasticsearch cluster.
```

Comment:
Should we link to the Stack Monitoring feature docs? This was it's clear that it's a Stack feature, not ECK feature.